src/main.rs

```diff
@@ -33,98 +34,110 @@ mod template;
 type Feature<'a> = &'a str;
 type PackageName<'a> = &'a str;
 type RootFeature<'a> = (PackageName<'a>, Feature<'a>);
+type Result<T> = std::result::Result<T, Error>;
 
 const VERSION_ATTRIBUTE_NAME: &str = "cargo2nixVersion";
 
 fn main() {
     let args: Vec<String> = std::env::args().collect();
     let args: Vec<&str> = args.iter().map(AsRef::as_ref).collect();
+    if let Err(err) = try_main(&args) {
+        eprint!(colorify!(red_bold: "error: "));
+        eprintln!("{}", err);
+        std::process::exit(1);
+    }
```

Comment:
non-issue: just wondering if we can simply write `fn main() -> Result<(), Error>`?

Reply:
As described in the first commit message (99871bc), the issue is that having `main()` return a `Result` is that it prints the `Debug` representation of the error instead of `Display`, which results in a poor experience for the end-user as a CLI tool.